improver/calibration/ensemble_calibration.py

```diff
@@ -562,114 +557,201 @@ def __repr__(self):
             self.max_iterations,
         )
 
-    def create_coefficients_cube(self, optimised_coeffs, historic_forecast):
-        """Create a cube for storing the coefficients computed using EMOS.
+    def _multiple_beta_coefficients(self, optimised_coeffs, historic_forecast):
+        """Re-structure the coefficients, if provided multiple beta values
+
+        Args:
+            optimised_coeffs (numpy.ndarray):
+                Order of coefficients is [alpha, beta, gamma, delta].
+            historic_forecast (iris.cube.Cube):
+                The cube containing the historic forecast.
+
+        Returns:
+            list:
+                List of coefficients containing a list of beta coefficients.
+                Order of coefficients is [alpha, beta, gamma, delta].
+
+        Raises:
+            ValueError: If the number of coefficients are not as expected when
+                multiple beta coefficients are provided.
+        """
+        coeff_names_without_beta = self.coeff_names[:]
+        coeff_names_without_beta.remove("beta")
+        if len(optimised_coeffs) != len(coeff_names_without_beta) + len(
+            historic_forecast.coord("realization").points
+        ):
+            msg = (
+                "The number of coefficients provided {} must equal the "
+                "alpha, gamma and delta coefficients plus the beta "
+                "coefficients where the number of beta coefficients equals "
+                "the number of realizations {}, when the predictor is "
+                "'realizations'.".format(
+                    optimised_coeffs, historic_forecast.coord("realization").points,
+                )
+            )
+            raise ValueError(msg)
+
+        return [
+            optimised_coeffs[0],
+            optimised_coeffs[1:-2],
+            optimised_coeffs[-2],
+            optimised_coeffs[-1],
+        ]
+
+    @staticmethod
+    def _set_attributes(historic_forecast):
+        """Set attributes for use on the EMOS coefficients cube.
+
+        Args:
+            historic_forecast (iris.cube.Cube)
+
+        Returns:
+            attributes (dict):
```

Comment:
I think the docstring style should be just `dict:` here.

Reply:
Altered.